src/main/scala/epic/constraints/LabeledSpanConstraints.scala

```diff
@@ -363,30 +354,14 @@ object LabeledSpanConstraints {
     // could avoid the allocation, but whatever.
     if(a.length < b.length) elementwiseMax(util.Arrays.copyOf(a, b.length), b)
     else if(b.length < a.length) elementwiseMax(a, util.Arrays.copyOf(b, a.length))
-    else {
-      val result = new Array[Int](a.length)
-      var i = 0
-      while(i < result.length) {
-        result(i) = math.max(a(i), b(i))
-        i += 1
-      }
-      result
-    }
+    else Array.fillWith[Int](a.length)(i => math.max(a(i), b(i)))
```

Comment:
why not Array.tabulate? Also, in general I'm a bit concerned about replacing while loops with HOFs. If it's a while loop, I was probably concerned about performance.